it/src/main/java/com/google/cloud/teleport/it/testcontainers/TestContainerResourceManager.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.testcontainers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+
+/** Interface for managing TestContainers resources in integration tests. */
+public abstract class TestContainerResourceManager<T extends GenericContainer<?>> {
+  private static final Logger LOG = LoggerFactory.getLogger(TestContainerResourceManager.class);
+
+  private final T container;
+  private final boolean usingStaticContainer;
```

Comment:
Giving some context on what is static container will he helpful. Maybe at class level?

Reply:
ack.